src/app/core/components/thread/thread.component.html

```diff
@@ -61,18 +61,21 @@
               tooltipPosition="top"
               tooltipStyleClass="alg-tooltip"
             >
-              <button
-                class="alg-button primary small"
-                [disabled]="!canSwitch"
-                pButton
-                (click)="changeThreadStatus()"
-                *ngIf="threadStatus && threadStatus.data"
-              >
-              <span *ngIf="threadStatus.data.open; else openThreadCaption" i18n>Close this thread</span>
-              <ng-template #openThreadCaption>
-                <span i18n>Open this thread</span>
-              </ng-template>
-              </button>
+              <ng-container *ngIf="threadInfo$ | async as threadInfo">
+                <button
+                  type="button"
+                  class="alg-button primary small"
+                  [disabled]="!canSwitch"
+                  pButton
+                  (click)="changeThreadStatus(!threadStatus.data.open, threadInfo.data)"
+                  *ngIf="threadStatus && threadStatus.data && threadInfo.data"
```

Comment:
From what I see, you do not user threadInfo$ but just the threadId$... if so, just use `this.threadService.threadId$,` directly, that's easier (not a fetch state).

Reply:
Fixed